java/vector/src/main/java/org/apache/arrow/vector/stream/MessageSerializer.java

```diff
@@ -120,6 +120,9 @@ public static ArrowBlock serialize(WriteChannel out, ArrowRecordBatch batch)
 
     long start = out.getCurrentPosition();
     int bodyLength = batch.computeBodyLength();
+    if (bodyLength % 8 != 0) {
+      bodyLength += 8 - (bodyLength % 8);
+    }
```

Comment:
I am curious why the body length would _not_ be a multiple of 8? I guess there buffers whose length is not padded to 8 bytes?  This seems problematic because it means that some buffers may not start on an aligned offset

Should we also ensure that schema messages are padded to an 8 byte multiple?